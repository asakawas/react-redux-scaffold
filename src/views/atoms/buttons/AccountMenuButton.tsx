import React, { FunctionComponent } from 'react';
import IconButton, { IconButtonProps } from '@material-ui/core/IconButton';
import Avatar from '@material-ui/core/Avatar';
import AccountCircle from '@material-ui/icons/AccountCircle';
import styled from 'styled-components';
import theme from '~/views/theme';

const StyledAvatar = styled(Avatar)`
  background-color: ${theme.palette.primary.light};
`;

/**
 * The type of props of AccountMenuButton.
 */
export type AccountMenuButtonProps = Readonly<{
  /** The event handler called when the menu button is clicked */
  onAccountMenuButtonClick: IconButtonProps['onClick'];
}>;

const AccountMenuButton: FunctionComponent<AccountMenuButtonProps> = ({
  onAccountMenuButtonClick,
}) => (
  <IconButton
    edge="end"
    onClick={onAccountMenuButtonClick}
    size="small"
    aria-label="account menu button"
  >
    <StyledAvatar>
      <AccountCircle color="action" />
    </StyledAvatar>
  </IconButton>
);

export default AccountMenuButton;
