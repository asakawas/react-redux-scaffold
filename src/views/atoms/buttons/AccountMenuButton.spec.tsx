import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import AccountMenuButton from './AccountMenuButton';

describe('AccountMenuButton', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<AccountMenuButton onAccountMenuButtonClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls the passed handler when it's clicked", () => {
    const handler = jest.fn();
    const renderResult = render(<AccountMenuButton onAccountMenuButtonClick={handler} />);

    const tree = renderResult.getByLabelText('account menu button');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(1);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(2);
  });
});
