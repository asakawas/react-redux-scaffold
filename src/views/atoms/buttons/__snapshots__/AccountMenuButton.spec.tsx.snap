// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountMenuButton is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #b3d6a4;
}

<button
    aria-label="account menu button"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiIconButton-edgeEnd"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle c0 MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorAction"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
